modules/network/src/main/java/org/apache/ignite/network/DefaultMessagingService.java

```diff
@@ -326,18 +326,25 @@ private void sendToSelf(NetworkMessage msg, @Nullable Long correlationId) {
      * @param obj Incoming message wrapper.
      */
     private void onMessage(InNetworkObject obj) {
-        if (isInNetworkThread()) {
-            inboundExecutor.execute(() -> {
-                try {
-                    onMessage(obj);
-                } catch (Throwable e) {
-                    logAndRethrowIfError(obj, e);
+        assert isInNetworkThread();
+
+        inboundExecutors.execute(obj.connectionIndex(), () -> {
+            long startedNanos = System.nanoTime();
+
+            try {
+                handleIncomingMessage(obj);
+            } catch (Throwable e) {
+                logAndRethrowIfError(obj, e);
+            } finally {
+                long tookMillis = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startedNanos);
+                if (tookMillis > 100) {
```

Comment:
Please have a look at https://github.com/apache/ignite-3/pull/2979. I believe that your additional logging duplicates already existing one.

Reply:
The timespan I log is a bit broader: it includes all the handlers (not just one) and also it includes unmarshalling. Probably having them both will allow to see the overhead of marshalling (and other handlers, if we ever use them).

So probably this might be useful too. WDYT?

Reply:
I'm fine with it